pandas/tseries/tests/test_timeseries.py

```diff
@@ -5393,6 +5394,73 @@ def test_day_not_in_month_ignore(self):
             '2015-04-31', errors='ignore', format="%Y-%m-%d"), '2015-04-31')
 
 
+class TestAstype(tm.TestCase):
+
+    def test_datetimeindex(self):
+        # GH 13149, GH 13209
+        idx = DatetimeIndex(['2016-05-16', 'NaT', NaT, np.NaN])
+
+        result = idx.astype(object)
+        expected = Index([Timestamp('2016-05-16')] + [NaT] * 3, dtype=object)
+        tm.assert_index_equal(result, expected)
+
+        result = idx.astype(int)
+        expected = Int64Index([1463356800000000000] +
+                              [-9223372036854775808] * 3, dtype=np.int64)
+        tm.assert_index_equal(result, expected)
+
+        result = idx.astype(str)
```

Comment:
An example from tests (nb. it failed `assert_index_equal` with pyhton2.7):

```python
In [2]: idx = DatetimeIndex(['2016-05-16', 'NaT', NaT, np.NaN])

In [3]: idx
Out[3]: DatetimeIndex(['2016-05-16', 'NaT', 'NaT', 'NaT'], dtype='datetime64[ns]', freq=None)

In [4]: idx2 = idx.astype(str)

In [5]: idx2
Out[5]: Index([u'2016-05-16', u'NaT', u'NaT', u'NaT'], dtype='object')

In [7]: type(idx2[0])
Out[7]: str

In [8]: type(idx2[1])
Out[8]: unicode
```

I think this `u('NaT')` may be responsible:
```python
    # class DatetimeIndex:
    def _format_native_types(self, na_rep=u('NaT'),
                             date_format=None, **kwargs):
```